### PR TITLE
Authn filter and Authz filter implementations with new syntax 

### PR DESCRIPTION
**Pull request by isurulucky**

### CODE REVIEW

File: stdlib/ballerina-http/src/main/ballerina/net.http.authadaptor/authn-filter.bal [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+            break;
+        }
+    }
+    if (lengthof authAnn == 0) {
+        // no annotation found for ballerina.auth:config
+        error err = {message:"no annotation for ballerina.auth:Config"};
+        return err;
+    }
+    var authConfig = <auth:AuthConfig> authAnn.value;
+    match authConfig {
```
Comment (afkham) [inline stdlib/ballerina-http/src/main/ballerina/net.http.authadaptor/authn-filter.bal]: You don't need match here. You can use ?=

File: stdlib/ballerina-http/src/main/ballerina/net.http.authadaptor/authz-filter.bal [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+        }
+        error err => {
+            // if not found in resource level, check in service level
+            scope = getAuthzAnnotation(internal:getServiceAnnotations(context.serviceType));
+            match scope {
+                string scopeVal => {
+                    return scopeVal;
+                }
+                error err => {
+                    // if the scope is still null, means authorization is not needed.
```
Comment (afkham) [inline stdlib/ballerina-http/src/main/ballerina/net.http.authadaptor/authz-filter.bal]: Isn't this a dangerous assumption? If there is an error, you are saing authz is not needed.

File: stdlib/ballerina-http/src/main/ballerina/net.http.authadaptor/authz-filter.bal [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+        }
+    }
+    if (lengthof authAnn == 0) {
+        // no annotation found for ballerina.auth:config
+        error err = {message:"no annotation for ballerina.auth:Config"};
+        return err;
+    }
+    var authConfig = <auth:AuthConfig> authAnn.value;
+    match authConfig {
+        auth:AuthConfig authConfiguration => {
```
Comment (afkham) [inline stdlib/ballerina-http/src/main/ballerina/net.http.authadaptor/authz-filter.bal]: You don't need match here. You can use ?=

File: tests/ballerina-test-integration/src/test/resources/auth/ballerina.conf [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+password="BA7816BF8F01CFEA414140DE5DAE2223B00361A396177A9CB410FF61F20015AD"
+groups="xyz"
+[user3]
+password="FADFEF49B40BF551A279F820BD863AC96AEBCBF39B4431DFF4F0D5CB62DD5303"
+
+# scopes
+[scope1]
+groups="xyz,pqr"
+[scope2]
+groups="lmn"
```
Comment (afkham) [inline tests/ballerina-test-integration/src/test/resources/auth/ballerina.conf]: Does this work with the new Config API change which expects the file to be in TOML?

Comment (isurulucky) [reply tests/ballerina-test-integration/src/test/resources/auth/ballerina.conf]: Yes, we have used this in other tests as well.

File: stdlib/ballerina-http/src/main/ballerina/net.http.authadaptor/authn-filter.bal (hunk 1 above)
Comment (isurulucky) [reply stdlib/ballerina-http/src/main/ballerina/net.http.authadaptor/authn-filter.bal]: fixed.

File: stdlib/ballerina-http/src/main/ballerina/net.http.authadaptor/authz-filter.bal (hunk 2 above)
Comment (isurulucky) [reply stdlib/ballerina-http/src/main/ballerina/net.http.authadaptor/authz-filter.bal]: changed the downstream method to return a null if no scopes are found and used a match statement.

File: stdlib/ballerina-http/src/main/ballerina/net.http.authadaptor/authz-filter.bal (hunk 3 above)
Comment (isurulucky) [reply stdlib/ballerina-http/src/main/ballerina/net.http.authadaptor/authz-filter.bal]: fixed.

**Pull request merged**